tensorflow/python/keras/backend.py

```diff
@@ -4313,6 +4313,12 @@ def binary_crossentropy(target, output, from_logits=False):
   if not from_logits:
     if (isinstance(output, (ops.EagerTensor, variables_module.Variable)) or
         output.op.type != 'Sigmoid'):
+      try:
+        target.get_shape().merge_with(output.get_shape())
+      except ValueError:
```

Comment:
Can we use assert_is_compatible_with instead of try catch?

Reply:
Thanks @pavithrasv. The PR has been updated with `assert_is_compatible_with`.